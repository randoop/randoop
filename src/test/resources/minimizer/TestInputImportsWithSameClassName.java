import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestInputImportsWithSameClassName {

  /* This test input uses ClassA belonging to package dir_a and ClassA belonging to
   * package dir_b. It would be nicer if the minimizer retained the
   * fully-qualified type name for both instances of ClassA.
   */

  @Test
  public void test1() throws Throwable {
    test.minimizer.dir_a.ClassA dirAObject = new test.minimizer.dir_a.ClassA();
    test.minimizer.dir_b.ClassA dirBObject = new test.minimizer.dir_b.ClassA();

    dirAObject.setId(1);
    dirBObject.setId(2);

    org.junit.Assert.assertFalse(dirAObject.getId() == dirBObject.getId());

    dirAObject.setId(100);
    dirBObject.setId(100);
    // Fails, should be true
    org.junit.Assert.assertFalse(dirAObject.getId() == dirBObject.getId());
  }
}
