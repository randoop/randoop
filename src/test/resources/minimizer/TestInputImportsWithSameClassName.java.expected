import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.runners.MethodSorters;
import test.minimizer.dir_a.ClassA;

@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class TestInputImportsWithSameClassNameMinimized {

    /* This test input uses ClassA belonging to package dir_a and ClassA belonging to
   * package dir_b. It would be nicer if the minimizer retained the
   * fully-qualified type name for both instances of ClassA.
   */
    @Test
    public void test1() throws Throwable {
        ClassA dirAObject = new ClassA();
        test.minimizer.dir_b.ClassA dirBObject = new test.minimizer.dir_b.ClassA();
        dirAObject.setId(100);
        dirBObject.setId(100);
        // Fails, should be true
        org.junit.Assert.assertFalse(dirAObject.getId() == dirBObject.getId());
    }
}
